### DISCUSSION TITLE
How to use FastAPI Dependency Injection features in places other than path operation functions? (In logging config to be precise)

### DISCUSSION THREAD

**Discussion by nilTheDev** [fastapi/fastapi#8487] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class CustomLoggerAdapter(logging.LoggerAdapter):
    """
        This overridden "process" method prepends the current
        user_id with all the log messages.
    """

    def process(self, msg, kwargs):
        from logic.auth.token_management import get_user_from_token
        from fastapi import Depends
        from data.models.users import Users
        def get_user(user: Users = Depends(get_user_from_token)):

            print(type(user))
            return user

        try:
            user = get_user()
        except:
            user = None

        return f'[user_id: {user.id if user else "null"}] {msg}', kwargs
```

### Description

* I am trying to add the currently logged-in `user_id` in all log messages. 

* I am using Python's `logging` library.

* I have created a `CustomLoggerAdapter` where the overridden `process` method would prepend the `user_id` in the log message. 

* I have to use FastAPI's DI system to get the current user.

* However, it seems FastAPI doesn't like its DI system to be used anywhere else other than the path-operation functions.

* So, is there any way I can use the DI system from the logging config-related codes?  If not then is there any you can think of to achieve this? 
 

 

 

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

3.9.10

**Comment by odiseo0** [comment]
That's not possible, `Depends` only works in FastAPI Endpoints

**Comment by rustam-ashurov-mcx** [comment]
I would give a chance to some DI libraries for such things. Dependency-injector works great with FatApi, but I guess there are other libraries
